plugins/reports/monographReport/MonographReportPlugin.php

```diff
@@ -0,0 +1,409 @@
+                    __('user.url') . ' (' . __('user.role.author') . ' ' . ($i + 1) . ')' => fn () => $this->getAuthor($i)?->getData('url'),
+                    __('user.biography') . ' (' . __('user.role.author') . ' ' . ($i + 1) . ')' => fn () => html_entity_decode(strip_tags($this->getAuthor($i)?->getLocalizedData('biography')))
+                ])
+                ->collapse(),
+            ...collect(range(0, $this->maxEditors - 1))
+                ->map(fn ($i) => [
+                    __('user.givenName') . ' (' . __('user.role.editor') . ' ' . ($i + 1) . ')' => fn () => $this->getEditor($i)?->getLocalizedGivenName(),
+                    __('user.familyName') . ' (' . __('user.role.editor') . ' ' . ($i + 1) . ')' => fn () => $this->getEditor($i)?->getLocalizedFamilyName(),
+                    __('user.orcid') . ' (' . __('user.role.editor') . ' ' . ($i + 1) . ')' => fn () => $this->getEditor($i)?->getData('orcid'),
+                    __('user.email') . ' (' . __('user.role.editor') . ' ' . ($i + 1) . ')' => fn () => $this->getEditor($i)?->getEmail(),
+                    ...collect(range(0, $this->maxDecisions - 1))
+                        ->map(fn ($j) => [
+                            __('manager.setup.editorDecision') . ' ' . ($j + 1) . ' (' . __('user.role.editor') . ' ' . ($i + 1) . ')' => fn () => $this->getDecisionMessage($this->getDecision($i, $j)?->getData('decision')),
+                            __('common.dateDecided') . ' ' . ($j + 1) . ' (' . __('user.role.editor') . ' ' . ($i + 1) . ')' => fn () => $this->getDecision($i, $j)?->getData('dateDecided')
+                        ])
+                        ->collapse()
+                ])
+                ->collapse()
+        ];
+    }
+
+    /**
+     * Retrieves a cached user
+     */
+    private function getUser(int $userId): ?User
+    {
+        static $users = [];
+        return $users[$userId] ??= Repo::user()->get($userId, true);
+    }
+
+    /**
+     * Retrieves a SplFileObject and sends HTTP headers to enforce the report download
+     */
+    private function createOutputStream(): SplFileObject
+    {
+        $acronym = PKPString::regexp_replace('/[^A-Za-z0-9 ]/', '', $this->press->getLocalizedAcronym());
+        $date = (new DateTimeImmutable())->format('Ymd');
+
+        // Prepare for UTF8-encoded CSV output.
+        header('content-type: text/comma-separated-values');
+        header("content-disposition: attachment; filename=monographs-{$acronym}-{$date}.csv");
+
+        $output = new SplFileObject('php://output', 'w');
+        // UTF-8 BOM to force the file to be read with the right encoding
+        $output->fwrite("\xEF\xBB\xBF");
+        return $output;
+    }
+
+    /**
+     * Retrieves the maximum amount of authors, editors and decisions that a submission may have
+     */
+    private function retrieveLimits(): void
```

Comment:
The original plugin is storing all the record set in memory, just to collect some aggregate data, this might lead to an undefined behavior, so I've added this method to pre-collect the "statistics".